added language template for cheatsheets, organized languages

iambibhas opened the pull request:
New template
---
Added new template for cheatsheets - `language`. Beside the usual `key` and `val`, this template also accepts `trn` which is used to show the transliterated version of the phrases i.e. the native words in latin/english characters, so that the non-native speakers can also use these language cheatsheets to speak most commonly used phrases. This is how it looks - 

![selection_029](https://cloud.githubusercontent.com/assets/357253/10289154/9863ab50-6bba-11e5-8fc1-fafe979046e8.png)

Example codes are here - https://github.com/iambibhas/zeroclickinfo-goodies/blob/0e7b7905bf1e0fdcaa57e4fadad7ae6f59d2989d/share/goodie/cheat_sheets/json/language/bengali.json

Reorganization
---
Also moved the language cheatsheets to a separate folders called `language` so they stay separated from rest of the cheatsheets and we can focus on adding more and more language cheatsheets.

I'll create an issue to move the existing language cheatsheets to the `language` template once this gets merged.

moollaza commented on share/goodie/cheat_sheets/language.handlebars:
```diff
@@ -0,0 +1,9 @@
+    <li class="text--primary">
+      <span class="cheatsheet__key">{{key}}</span>
+      <span class="cheatsheet__value text--secondary">{{val}}</span>
+      <span class="cheatsheet__translit text--secondary">{{trn}}</span>
```
I believe we'll want this to be optional, depending on whether `trn` is defined? (If we're going to use this for all "language" cheat sheets)

I'm not a huge fan of `trn` because it's not as obvious as `key` and `val` but I guess it works for now.

iambibhas replied:
I also dont like `trn` that much, but just wanted to have some uniformity between the the key names. I first named if `translit`, but seeing that the others are `key` and `val`, renamed it. I'll make it optional.

moollaza commented on share/goodie/cheat_sheets/cheat_sheets.css:
```diff
@@ -102,6 +102,41 @@
+  display: block;
+}
+
+.zci--cheat_sheets .language li .cheatsheet__translit {
```
I think we set a bad example using `<span>` in the reference template. We should just use `<div>` so we don't need CSS setting it to `display: block`

iambibhas replied:
Yeah, thought so too. Will do that.

iambibhas commented:
@moollaza should I update the reference template to use `div` too?

iambibhas commented:
@moollaza seems like forcing the `val` to use `font-family: monospace` renders the non-latin languages properly. Here is a before-after comparison of Bengali. Shows the different in fused letters - 

![goodie-font](https://cloud.githubusercontent.com/assets/357253/10304992/1382a388-6c3b-11e5-847b-cca14e33fa45.png)

Although that forces other languages to show up monospaced - 

![selection_034](https://cloud.githubusercontent.com/assets/357253/10305006/2d40298a-6c3b-11e5-9d5e-7091fe9e3fb7.png)

What to do?

moollaza commented:
@iambibhas how does the site look to you, when you switch the preferred language to Bengali? It seems like things are a little clearer when the font is slightly larger, so perhaps we'll need to bump increase the font size a bit, or reduce the font weight. Let's try that first.

As a last resort, I suppose we can apply the monspace font-face when `trn` is defined, meaning we're dealing with a non latin-based alphabet

AhmedBenmoussa commented:
Great work @iambibhas. Really great. can't wait to try it.

iambibhas commented:
> how does the site look to you, when you switch the preferred language to Bengali?

Same problem with fused letters.

![selection_036](https://cloud.githubusercontent.com/assets/357253/10316439/e13f9eca-6c7c-11e5-8cc7-f6e588407015.png)

If I make the fonts monospaced, then only they look proper. Maybe we can try the same with some other language? Maybe that Arabic cheatsheet PR we have pending. Just ask the dev to try it out.

> As a last resort, I suppose we can apply the monspace font-face when trn is defined, meaning we're dealing with a non latin-based alphabet

@moollaza I think this is what we should do.

moollaza commented:
@iambibhas what browser/OS are you using? Are the font's being rendered with our DDG_Proxima_Nova font? Things look much clearer to me on OSX Firefox, Safari
<img width="1422" alt="_" src="https://cloud.githubusercontent.com/assets/873785/10318191/50fe8718-6c36-11e5-95c3-be3f9232a718.png">
, Chrome

iambibhas commented:
@moollaza Firefox and Chromium, Ubuntu. It seems like that Proxima Nova font is being used. But still the fonts are not being rendered properly. Do you have any clue?

moollaza commented:
@iambibhas I'm not sure what the problem could be. Do you have any other computers/devices you could test on to see if it's just your computer that has the problem?

iambibhas commented:
@moollaza it works on other devices. Maybe something wrong with my comp somehow. Will figure that out. This is good to go otherwise. Should I make `reference` template use `div` too?

iambibhas commented:
Fixed the issue on my comp. Turns out the font family `FreeFont` was being used as default for Bengali texts and it didn't handle the conjuncts well. Removed it. Now everything is good. :)

moollaza commented:
>Fixed the issue on my comp. Turns out the font family FreeFont was being used as default for Bengali texts and it didn't handle the conjuncts well. Removed it. Now everything is good. :)

Awesome! I'm glad to hear you figured this out :+1: 

> Should I make reference template use div too? 

I think it's OK for now. We can review the HTML/CSS for all templates and simplify things if necessary.

@iambibhas I need you to add the actual Bengali Cheat Sheet in a separate PR though, after this one is submitted so the connection to the IA Page will work properly.

moollaza commented:
@iambibhas ping!

iambibhas commented:
@moollaza done. will send separate PR for the bengali cheatsheet once this is merged.

moollaza commented:
@iambibhas Thanks! This LGTM :+1:
